pkg/dogstatsd/listeners/packet_buffer.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package listeners
+
+import (
+	"sync"
+	"time"
+)
+
+type packetBuffer struct {
```

Comment:
I'll be very picky on this one, please ignore it if you feel like it: it's not only a packet buffer but also a packet sender (flushing to `outputChannel`). It may be useful to reflect it in its name or more simply in a comment here if you want to avoid a long name.

Reply:
I also don't find this name very descriptive but I can't think of a name describing it better, any ideas ?

Reply:
I think I would've go for `bufferedPacketSender` to be descriptive but that's a bit long... Inspired from https://golang.org/pkg/bufio/ it could be `bufPacketSender` but well, doesn't look very good either... 🤷‍♂️ 
After 10 minutes looking for a name I've a tendency of adding a small comment on the struct even if private 😆

Reply:
Not sure `sender` is the right word either. I'll add a few comments to make sure no one gets confused